packages/launch/src/focused-launch/index.tsx

```diff
@@ -39,6 +39,17 @@ const FocusedLaunch: React.FunctionComponent = () => {
 		}
 	}, [ selectedDomain, domainSuggestionFromCart, setDomain ] );
 
+	// If there is no selected plan, but there is a plan in cart,
+	// set the plan from cart as the selected plan.
+	const planFromCart = usePlanFromCart();
+	const selectedPlan = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedPlan() );
+	const { setPlan } = useDispatch( LAUNCH_STORE );
+	React.useEffect( () => {
+		if ( ! selectedPlan && planFromCart ) {
+			setPlan( planFromCart );
+		}
+	}, [ selectedPlan, planFromCart, setPlan ] );
```

Comment:
👍 
Tested all the possibilities including selecting a paid domain while having Free plan selected. Works like magic 🧙